ksonnet-gen/kubespec/new.go

```diff
@@ -0,0 +1,90 @@
+package kubespec
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var (
+	regexes = []*regexp.Regexp{
+		// Core API, pre-1.8 Kubernetes OR non-Kubernetes codebase APIs
+		regexp.MustCompile(`io\.k8s\.(?P<codebase>\S+)\.pkg\.api\.(?P<version>\S+)\.(?P<kind>\S+)`),
+		// Core API, 1.8+ Kubernetes
+		regexp.MustCompile(`io\.k8s\.api\.(?P<packageType>core)\.(?P<version>\S+)\.(?P<kind>\S+)`),
+		// Other APIs, pre-1.8 Kubernetes OR non-Kubernetes codebase APIs
+		regexp.MustCompile(`io\.k8s\.(?P<codebase>\S+)\.pkg\.(?P<packageType>apis)\.(?P<group>\S+)\.(?P<version>\S+)\.(?P<kind>\S+)`),
+		// Other APIs, 1.8+ Kubernetes
+		regexp.MustCompile(`io\.k8s\.api\.(?P<group>\S+)\.(?P<version>\S+)\.(?P<kind>\S+)`),
+		// Util packageType
+		regexp.MustCompile(`io\.k8s\.(?P<codebase>\S+)\.pkg\.(?P<packageType>util)\.(?P<version>\S+)\.(?P<kind>\S+)`),
+		// Version packageType
+		regexp.MustCompile(`io\.k8s\.(?P<codebase>\S+)\.pkg\.(?P<packageType>version)\.(?P<kind>\S+)`),
+		// Runtime packageType
+		regexp.MustCompile(`io\.k8s\.(?P<codebase>\S+)\.pkg\.(?P<packageType>runtime)\.(?P<kind>\S+)`),
+	}
+)
+
+var (
+	packageTypeMap = map[string]Package{
+		"core":    Core,
+		"apis":    APIs,
+		"util":    Util,
+		"runtime": Runtime,
+		"version": Version,
+	}
+)
+
+type description struct {
```

Comment:
A few things.

1. It seems like `description#Validate` is not used?
2. It seems like actually `describeDefinition` is really just doing what `DefinitionName#Parse` should be doing? Should we just move this code there?
3. If I'm missing something in (2) and this needs to stay in its own file, do you mind if we call this file something other than `new.go` (and also rename `description` to something else, since it's really actually a `DefinitionName`)? :) (I suspect this was a placeholder name.)